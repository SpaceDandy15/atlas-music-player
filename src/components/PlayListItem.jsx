import React from "react";

export default function PlayListItem({
  title = "Song Title",
  artist = "Artist",
  length = "3:45",
  selected = false,
  onClick,
}) {
  return (
    <div
      className={`flex justify-between items-center px-4 py-2 cursor-pointer transition-colors
        ${selected
          ? "bg-[var(--color-tahiti)] text-white"
          : "hover:bg-[var(--color-bermuda)] hover:text-black group"
      }`}
      onClick={onClick}
    >
      <div className="flex flex-col">
        <p className="text-sm font-medium text-white group-hover:text-black">{title}</p>
        <p className="text-xs text-white group-hover:text-black">{artist}</p>
      </div>
      <span className="text-xs text-white group-hover:text-black">{length}</span>
    </div>
  );
}
