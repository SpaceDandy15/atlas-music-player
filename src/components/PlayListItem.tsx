import React from "react";

type PlayListItemProps = {
  title?: string;
  artist?: string;
  length?: string;
  selected?: boolean;
  onClick?: () => void;
};

export default function PlayListItem({
  title = "Song Title",
  artist = "Artist",
  length = "3:45",
  selected = false,
  onClick,
}: PlayListItemProps) {
  return (
    <div
      className={`flex justify-between items-center px-4 py-2 cursor-pointer transition-colors
        ${selected
          ? "bg-[var(--color-tahiti)] text-white"
          : "hover:bg-[var(--color-bermuda)] hover:text-black group"
      }`}
      onClick={onClick}
    >
      <div className="flex flex-col">
        <p className="text-sm font-medium text-white group-hover:text-black">{title}</p>
        <p className="text-xs text-white group-hover:text-black">{artist}</p>
      </div>
      <span className="text-xs text-white group-hover:text-black">{length}</span>
    </div>
  );
}
