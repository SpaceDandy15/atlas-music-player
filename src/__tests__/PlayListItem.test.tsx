import React from "react";
import { describe, it, expect, vi } from "vitest";
import renderer, { act } from "react-test-renderer";
import PlayListItem from "../components/PlayListItem";

describe("PlayListItem Component", () => {
  it("renders correctly with default props", () => {
    // Snapshot test with default values
    const tree = renderer.create(<PlayListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when selected with highlight color", () => {
    // Snapshot test with selected + custom highlightColor
    const tree = renderer
      .create(<PlayListItem selected highlightColor="red" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onClick when clicked", () => {
    // Mock click handler
    const handleClick = vi.fn();

    let component: renderer.ReactTestRenderer;
    act(() => {
      component = renderer.create(
        <PlayListItem title="Clickable Song" onClick={handleClick} />
      );
    });

    // Get the root instance *before* calling .toJSON()
    const instance = component!.root;

    act(() => {
      instance.props.onClick();
    });

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
