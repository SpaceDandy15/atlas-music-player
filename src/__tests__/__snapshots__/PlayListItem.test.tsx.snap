// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Component > renders correctly when selected with highlight color 1`] = `null`;

exports[`PlayListItem Component > renders correctly with default props 1`] = `null`;
